runbotics/runbotics-scheduler/src/scheduler-database/process-instance/process-instance.service.ts

```diff
@@ -0,0 +1,190 @@
+            ...specs,
+        };
+
+        options.relations = RELATIONS;
+        options.where = {
+            ...options.where,
+            process: {
+                tenantId: user.tenantId,
+            },
+        };
+
+        const processInstancePage = await getPage(
+            this.processInstanceRepository,
+            options
+        );
+
+        return processInstancePage;
+    }
+
+    async getOne(id: ProcessInstance['id'], user: UserEntity) {
+        const processInstance = await this.processInstanceRepository
+            .findOneOrFail({
+                where: {
+                    id,
+                    process: {
+                        tenantId: user.tenantId,
+                    },
+                },
+                relations: RELATIONS,
+            })
+            .catch(() => {
+                throw new NotFoundException(
+                    `Could not find process instance with id ${id}`
+                );
+            });
+
+        return processInstance;
+    }
+
+    async getSubprocesses(
+        id: ProcessInstance['id'],
+        user: UserEntity,
+        paging: Paging
+    ): Promise<Page<ProcessInstance>> {
+        const options: FindManyOptions<ProcessInstance> = {
+            ...paging,
+        };
+
+        options.relations = RELATIONS;
+        options.where = [
+            {
+                process: {
+                    tenantId: user.tenantId,
+                },
+                parentProcessInstanceId: id,
+            },
+            {
+                process: {
+                    tenantId: user.tenantId,
+                },
+                parentProcessInstanceId: IsNull(),
+                rootProcessInstanceId: id,
+            },
+        ];
+
+        const subprocessesPage = await getPage(
+            this.processInstanceRepository,
+            options
+        );
+
+        return subprocessesPage;
+    }
+
+    async update(
+        id: ProcessInstance['id'],
+        processInstanceDto: UpdateProcessInstanceDto
+    ) {
+        await this.processInstanceRepository
+            .findOneByOrFail({ id })
+            .catch(() => {
+                throw new BadRequestException(
+                    `Cannot find process instance with provided id ${id}`
+                );
+            });
+
+        return this.processInstanceRepository.update(id, processInstanceDto);
+    }
+
+    findOneById(id: ProcessInstance['id']) {
+        return this.processInstanceRepository.findOne({
+            where: { id },
+            relations: RELATIONS,
+        });
+    }
+
+    findAllByBotId(id: IBot['id'], tenantId: Tenant['id']) {
+        return this.processInstanceRepository.find({
+            where: {
+                bot: { id },
+                process: { tenantId },
+            },
+            relations: RELATIONS,
+        });
+    }
+
+    findAllActiveByBotId(id: IBot['id']) {
+        const date = new Date();
+        date.setDate(date.getDate() - 7);
```

Comment:
I'd be great to make this 7 or even the whole 'date.getDate() - 7' const, uppercase variable describing what is this subtraction for

Reply:
still don't seeing this